libs/libarchfpga/src/read_xml_arch_file.cpp

```diff
@@ -1251,18 +1263,24 @@ static void ProcessPb_Type(vtr::string_internment* strings, pugi::xml_node Paren
 
     /* Count stats on the number of each type of pin */
     pb_type->num_clock_pins = pb_type->num_input_pins = pb_type->num_output_pins = 0;
-    for (i = 0; i < pb_type->num_ports; i++) {
-        if (pb_type->ports[i].type == IN_PORT
-            && pb_type->ports[i].is_clock == false) {
-            pb_type->num_input_pins += pb_type->ports[i].num_pins;
-        } else if (pb_type->ports[i].type == OUT_PORT) {
-            pb_type->num_output_pins += pb_type->ports[i].num_pins;
-        } else {
-            VTR_ASSERT(pb_type->ports[i].is_clock
-                       && pb_type->ports[i].type == IN_PORT);
-            pb_type->num_clock_pins += pb_type->ports[i].num_pins;
+    t_port* pb_type_ports = pb_type->ports;
+    int l = 0;
+    do {
```

Comment:
Need a comment on why you do this.  (two types of latch class etc.)

Reply:
Added comments